ingest/scripts/get_loculus_depositions.py

```diff
@@ -0,0 +1,161 @@
+import logging
+import os
+import re
+from dataclasses import dataclass
+
+import click
+import yaml
+from psycopg2.extras import RealDictCursor
+from psycopg2.pool import SimpleConnectionPool
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(
+    encoding="utf-8",
+    level=logging.DEBUG,
+    format="%(asctime)s %(levelname)8s (%(filename)20s:%(lineno)4d) - %(message)s ",
+    datefmt="%H:%M:%S",
+)
+
+
+@dataclass
+class Config:
+    segmented: bool
+    db_password: str
+    db_username: str
+    db_host: str
+
+
+def convert_jdbc_to_psycopg2(jdbc_url):
+    jdbc_pattern = r"jdbc:postgresql://(?P<host>[^:/]+)(?::(?P<port>\d+))?/(?P<dbname>[^?]+)"
+
+    match = re.match(jdbc_pattern, jdbc_url)
+
+    if not match:
+        msg = "Invalid JDBC URL format."
+        raise ValueError(msg)
+
+    host = match.group("host")
+    port = match.group("port") or "5432"  # Default to 5432 if no port is provided
+    dbname = match.group("dbname")
+
+    return f"postgresql://{host}:{port}/{dbname}"
+
+
+def db_init(
+    db_password_default: str, db_username_default: str, db_url_default: str
+) -> SimpleConnectionPool:
+    db_password = os.getenv("DB_PASSWORD")
+    if not db_password:
+        db_password = db_password_default
+
+    db_username = os.getenv("DB_USERNAME")
+    if not db_username:
+        db_username = db_username_default
+
+    db_url = os.getenv("DB_URL")
+    if not db_url:
+        db_url = db_url_default
+
+    db_dsn = convert_jdbc_to_psycopg2(db_url) + "?options=-c%20search_path%3Dena_deposition_schema"
+    return SimpleConnectionPool(
+        minconn=1,
+        maxconn=2,  # max 7*2 connections to db allowed
+        user=db_username,
+        password=db_password,
+        dsn=db_dsn,
+    )
+
+
+def get_bio_sample_accessions(db_conn_pool: SimpleConnectionPool) -> dict[str, str]:
+    con = db_conn_pool.getconn()
+    try:
+        with con, con.cursor(cursor_factory=RealDictCursor) as cur:
+            # Result is a jsonb column
+            query = "SELECT accession, result FROM sample_table WHERE STATUS = 'SUBMITTED'"
+
+            cur.execute(query)
+
+            results = cur.fetchall()
+    finally:
+        db_conn_pool.putconn(con)
+
+    return {result["accession"]: result["result"]["biosample_accession"] for result in results}
+
+
+def get_insdc_accessions(db_conn_pool: SimpleConnectionPool) -> dict[str, str]:
+    con = db_conn_pool.getconn()
+    try:
+        with con, con.cursor(cursor_factory=RealDictCursor) as cur:
+            # Result is a jsonb column
+            query = "SELECT accession, result FROM assembly_table WHERE STATUS = 'SUBMITTED'"
```

Comment:
query = "SELECT accession, result FROM assembly_table WHERE STATUS IN ('SUBMITTED', 'WAITING')"
```suggestion
            query = "SELECT accession, result FROM assembly_table WHERE STATUS IN ('SUBMITTED', 'WAITING')"
```